File: extractor-appender/appender-sql/src/main/java/ru/mydesignstudio/database/metadata/extractor/database/trigger/TriggerExtractor.java
package ru.mydesignstudio.database.metadata.extractor.database.trigger;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import ru.mydesignstudio.database.metadata.extractor.database.ExtractHelper;
import ru.mydesignstudio.database.metadata.extractor.database.TableMetadataAppender;
import ru.mydesignstudio.database.metadata.extractor.extract.result.TableMetadata;
import ru.mydesignstudio.database.metadata.extractor.extract.result.TriggerModel;
import ru.mydesignstudio.database.metadata.extractor.resource.StringResource;

@Slf4j
@Component
public class TriggerExtractor implements TableMetadataAppender {
  @StringResource("classpath:sql/extract_triggers.sql")
  private String extractQuery;

  @Autowired
  private ExtractHelper helper;

  @Override
  public void append(TableMetadata metadata, String schemaName, String tableName) {
    log.info("Extracting triggers for table {} in schema {}", tableName, schemaName);

    metadata.setTriggers(helper.extract(extractQuery, new Object[] { tableName, schemaName }, TriggerModel.class));

    log.debug("Extracted {} triggers", metadata.getTriggers().size());
  }
}

File: extractor-source/source-oracle/src/main/kotlin/ru/mydesignstudio/database/metadata/extractor/table/extractor/TriggerExtractor.kt
package ru.mydesignstudio.database.metadata.extractor.table.extractor

import org.springframework.stereotype.Component
import ru.mydesignstudio.database.metadata.extractor.ExtractorBase
import ru.mydesignstudio.database.metadata.extractor.connection.DataSourceProvider
import ru.mydesignstudio.database.metadata.extractor.extract.helper.QueryProvider
import ru.mydesignstudio.database.metadata.extractor.extract.parameters.source.MetadataObject
import ru.mydesignstudio.database.metadata.extractor.extract.parameters.source.MetadataSchema
import ru.mydesignstudio.database.metadata.extractor.extract.parameters.source.MetadataSourceConnection
import ru.mydesignstudio.database.metadata.extractor.extract.result.TriggerModel

@Component
class TriggerExtractor (
        private val dataSourceProvider: DataSourceProvider,
        private val queryProvider: QueryProvider) : ExtractorBase() {
    fun extract(metadata: MetadataObject, schema: MetadataSchema, connection: MetadataSourceConnection): Collection<TriggerModel> {
        val query = queryProvider.provide("classpath:sql/extract_triggers.sql")
        val dataSource = dataSourceProvider.provide(connection)
        return extract(dataSource, query, listOf(metadata.name, schema.name)) { rs ->
            TriggerModel(
                    triggerSchemaName = getString(rs, "trigger_schema_name"),
                    triggerName = getString(rs, "trigger_name"),
                    triggerType = getString(rs, "trigger_type"),
                    triggeringEvent = getString(rs, "triggering_event"),
                    schemaName = getString(rs, "schema_name"),
                    objectName = getString(rs, "object_name"),
                    objectType = getString(rs, "object_type"),
                    status = getString(rs, "status"),
                    script = getString(rs, "script")
            )
        }
    }
}